### PR TITLE
Rawcam

### PR DESCRIPTION
**Pull request by audergon**

### CODE REVIEW

**Comment by 6by9** [comment]
Sorry, I have reservations about merging raspiraw even if the conflicts are resolved.
(a) having it present in this repo implies a level of support from Pi Towers. As noted in the headers, we can not support the register sets due to NDAs.
(b) The error handling in the app needs to be cleaned up
(c) The handling of GPIOs and I2C bus pinmuxing needs to be made totally generic. There are other changes in the pipeline that will conflict, or at duplicate, what is done here.

@popcornmix @JamesH65 What are your views if there is a demand for this app to be centrally maintained? A new repo, or at least a new directory within host_applications/linux/apps/? Or do we delay until V4L2 is sorted and push things that way?
In many ways raspiraw is easier for developing register sets than V4L2 as it is all done in userspace instead of having to rebuild kernel drivers, so I do see a case for having both available.

File: interface/mmal/core/mmal_logging.c [hunk 1]
```diff
@@ -28,7 +28,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "core/mmal_core_private.h"
 
 VCOS_LOG_CAT_T mmal_log_category;
-static VCOS_LOG_LEVEL_T mmal_log_level = VCOS_LOG_ERROR;
+static VCOS_LOG_LEVEL_T mmal_log_level = VCOS_LOG_TRACE;
```
Comment (JamesH65) [inline interface/mmal/core/mmal_logging.c]: Is this change intended?

File: interface/mmal/mmal_encodings.h [hunk 2]
```diff
@@ -1,29 +1,10 @@
...
-(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
-ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
-SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-*/
+=============================================================================*/
+/** \file
+ * Multi-Media Abstraction Layer - Definition of some standard encodings.
+ */
```
Comment (JamesH65) [inline interface/mmal/mmal_encodings.h]: Is this change intended?

**Comment by 6by9** [comment]
@JamesH65 This is a pull request by @audergon against my rawcam branch. Expect all sorts of oddball stuff in there.

As we had discussed briefly the other week, there is a question mark over whether we want this merged into the main userland (particularly as it requires several configuration changes to run), or if it is better to sort out a new repo for it. I think I'd prefer the latter, so I may look at splitting it off into a new repo of my own. Whether it then gets merged into a raspberrypi repo is a different question.

**Comment by JamesH65** [comment]
Closing. At this time we don't want this merged to our tree.

**Comment by 6by9** [comment]
For info, raspiraw has now been split out from being a branch of userland to an independent repo - https://github.com/6by9/raspiraw
I'll talk with those responsible for the main raspberrypi github account to see if they want to merge it there.
